matchmaking.py

```diff
@@ -146,6 +146,20 @@ def update_user_profile(self) -> None:
             except Exception:
                 pass
 
+    def get_weights(self, online_bots: list[USER_PROFILE_TYPE], rating_preference: str, min_rating: int, max_rating: int,
+                    game_type: str) -> list[int]:
+        """Get the weight for each bot. A higher weights means the bot is more likely to get challenged."""
+        if rating_preference == "high":
+            reduce_ratings_by = min(min_rating - (max_rating - min_rating), min_rating - 1)
+            weights = [bot.get("perfs", {}).get(game_type, {}).get("rating", 0) - reduce_ratings_by for bot in online_bots]
+        elif rating_preference == "low":
+            reduce_ratings_by = min(min_rating - (max_rating - min_rating), min_rating - 1)
+            weights = [(max_rating - bot.get("perfs", {}).get(game_type, {}).get("rating", 0)) - reduce_ratings_by
+                       for bot in online_bots]

```

Comment:
This formula isn't correct for `"low"`. It can result in negative weights. The correct formula and usage is
```python
reduce_ratings_by = max(max_rating - (min_rating - max_rating), max_rating + 1)
# or, reduce_ratings_by = max(2*max_rating - min_rating, max_rating + 1)
weights = [(reduce_ratings_by - bot.get("perfs", {}).get(game_type, {}).get("rating", 0))
           for bot in online_bots]
```
The variables `max_rating` and `min_rating` switch roles, `min()` becomes `max()`, and the order of subtraction is reversed.